### PR TITLE
Added solution for MatrixOperationTest

### PR DESCRIPTION
**Pull request by Nokafan**

### CODE REVIEW

File: src/main/java/core/basesyntax/MatrixOperation.java [hunk 1]
```diff
@@ -6,18 +6,60 @@
  */
 public class MatrixOperation {
     public int[] getDiagonal(int[][] matrix) {
-        return null;
+        if (matrix.length == 0) {
```
Comment (RomanDubovskyi) [inline src/main/java/core/basesyntax/MatrixOperation.java]: how is this check different from this row:
int[] resultArray = new int[matrix.length];

File: src/main/java/core/basesyntax/MatrixOperation.java [hunk 2]
```diff
@@ -6,18 +6,60 @@
...
+            return new int[0];
+        }
+        if (matrix.length == 1) {
```
Comment (RomanDubovskyi) [inline src/main/java/core/basesyntax/MatrixOperation.java]: same, these checks seem to be redundant

File: src/main/java/core/basesyntax/MatrixOperation.java [hunk 3]
```diff
@@ -6,18 +6,60 @@
...
+        }
+        return resultArray;
     }
 
     public int[] getCounterDiagonal(int[][] matrix) {
-        return null;
+        if (matrix.length == 0) {
+            return new int[0];
+        }
+        if (matrix.length == 1) {
```
Comment (RomanDubovskyi) [inline src/main/java/core/basesyntax/MatrixOperation.java]: same here

File: src/main/java/core/basesyntax/MatrixOperation.java [hunk 4]
```diff
@@ -6,18 +6,60 @@
...
+        }
+        return resultArray;
     }
 
     public int getMin(int[][] matrix) {
-        return 0;
+        if (matrix.length == 0) {
+            return 0;
+        }
+        int result = Integer.MAX_VALUE;
```
Comment (RomanDubovskyi) [inline src/main/java/core/basesyntax/MatrixOperation.java]: you don't need constant here, you may just choose assign result to first matrix element

File: src/main/java/core/basesyntax/MatrixOperation.java [hunk 5]
```diff
@@ -6,18 +6,60 @@
...
+        }
+        return result;
     }
 
     public int getMax(int[][] matrix) {
-        return 0;
+        if (matrix.length == 0) {
+            return 0;
+        }
+        int result = Integer.MIN_VALUE;
```
Comment (RomanDubovskyi) [inline src/main/java/core/basesyntax/MatrixOperation.java]: same here

File: src/main/java/core/basesyntax/MatrixOperation.java (hunk 1 above)
Comment (Nokafan) [reply src/main/java/core/basesyntax/MatrixOperation.java]: Fixed.

File: src/main/java/core/basesyntax/MatrixOperation.java (hunk 2 above)
Comment (Nokafan) [reply src/main/java/core/basesyntax/MatrixOperation.java]: Fixed

File: src/main/java/core/basesyntax/MatrixOperation.java (hunk 3 above)
Comment (Nokafan) [reply src/main/java/core/basesyntax/MatrixOperation.java]: Fixed

File: src/main/java/core/basesyntax/MatrixOperation.java (hunk 4 above)
Comment (Nokafan) [reply src/main/java/core/basesyntax/MatrixOperation.java]: Fixed as well.

File: src/main/java/core/basesyntax/MatrixOperation.java (hunk 5 above)
Comment (Nokafan) [reply src/main/java/core/basesyntax/MatrixOperation.java]: Fixed :)

**Review by boroda4436 — APPROVED** [review]

**Pull request closed without merging**